snmp/datadog_checks/snmp/snmp.py

```diff
@@ -153,6 +153,7 @@ def discover_instances(self):
                 else:
                     for profile in profiles:
                         host_config.refresh_with_profile(self.profiles[profile], self.warning, self.log)
+                    host_config.add_profile_tag(profile)
```

Comment:
This is a bit jarring — at first I thought the indent level was wrong. Maybe let's make the intent clearer here?

```suggestion
                    most_specific_profile = profiles[-1]
                    host_config.add_profile_tag(most_specific_profile)
```